app/tests/unit/routes/v1/status.spec.js

```diff
@@ -4,66 +4,145 @@ const request = require('supertest');
-              status: 'completed',
-              timestamp: 1571679722653
-            },
-            {
-              description: null,
-              status: 'delivered',
-              timestamp: 1571679722622
-            },
-            {
-              description: null,
-              status: 'processing',
-              timestamp: 1571679722044
-            },
-            {
-              description: null,
-              status: 'enqueued',
-              timestamp: 1571679721991
-            },
-            {
-              description: null,
-              status: 'accepted',
-              timestamp: 1571679721833
-            }
-          ],
-          tag: 'tag',
-          txId: '00000000-0000-0000-0000-000000000000',
-          updatedTimestamp: 1571679722674
-        };
-      }
-    };
+    expect(response.statusCode).toBe(200);
+    expect(response.body).toBeTruthy();
+    expect(Array.isArray(response.body)).toBeTruthy();
+    expect(response.body[0].createdTimestamp).toBe(1571679721833);
+    expect(response.body[0].delayTS).toBe(0);
+    expect(response.body[0].msgId).toMatch(id);
+    expect(response.body[0].status).toMatch('completed');
+    expect(response.body[0].tag).toBeTruthy();
+    expect(response.body[0].txId).toBeTruthy();
+    expect(response.body[0].updatedTimestamp).toBe(1571679722674);
+  });
+
+  it('should respond with an empty array if nothing was found', async () => {
+    ChesService.prototype.findStatuses.mockResolvedValue([]);
+
+    const response = await request(app).get(`${basePath}`).query({
+      fields: 'createdTimestamp,delayTS,updatedTimestamp',
+      msgId: '00000000-0000-0000-0000-000000000000',
+      status: 'completed',
+      tag: 'tag',
+      txId: '00000000-0000-0000-0000-000000000000'
+    });
+
+    expect(response.statusCode).toBe(200);
+    expect(response.body).toBeTruthy();
+    expect(Array.isArray(response.body)).toBeTruthy();
+    expect(response.body.length).toBe(0);
+  });
+
+  it('should respond with an internal server error', async () => {
+    const errorMsg = 'error';
+    ChesService.prototype.findStatuses.mockImplementation(() => {
+      throw new Error(errorMsg);
+    });
+
+    const response = await request(app).get(`${basePath}`).query({
+      fields: 'createdTimestamp,delayTS,updatedTimestamp',
+      msgId: '00000000-0000-0000-0000-000000000000',
+      status: 'completed',
+      tag: 'tag',
+      txId: '00000000-0000-0000-0000-000000000000'
+    });
+
+    expect(response.statusCode).toBe(500);
+    expect(response.body.title).toMatch('Internal Server Error');
+    expect(response.body.details).toMatch(errorMsg);
+  });
+
+  it('should respond with a validation error', async () => {
+    const response = await request(app).get(`${basePath}`);
+
+    expect(response.statusCode).toBe(422);
+    expect(response.body).toBeTruthy();
+    expect(response.body.detail).toMatch('Validation failed');
+    expect(response.body.errors).toHaveLength(1);
```

Comment:
Should we test which validation error is actually returned?

Reply:
Not extremely necessary here but is good to do to be consistent. Implemented.